packages/react-components/react-combobox/src/utils/useComboboxPopup.ts

```diff
@@ -0,0 +1,22 @@
+import { resolvePositioningShorthand, usePositioning } from '@fluentui/react-positioning';
+import { useMergedRefs } from '@fluentui/react-utilities';
+import type { ComboboxBaseProps } from '../ComboboxBase/ComboboxBase.types';
+import type { ComboboxState } from '../components/Combobox/Combobox.types';
+
+export const useComboboxPopup = (props: ComboboxBaseProps, state: ComboboxState) => {
+  const { positioning } = props;
+
+  // popper options
+  const popperOptions = {
+    position: 'below' as const,
+    align: 'start' as const,
+    offset: { crossAxis: 0, mainAxis: 2 },
+    ...resolvePositioningShorthand(positioning),
+  };
+
+  const { targetRef, containerRef } = usePositioning(popperOptions);
+
+  state.listbox.ref = useMergedRefs(state.listbox.ref, containerRef);
+
+  state.button.ref = useMergedRefs(state.button.ref, targetRef);
```

Comment:
nit: seems a bit inconsistent that you return state in other hooks in this PR and mutate it here

Reply:
true, originally I had the other hooks mutate state too 😅. I'll update this one to take/return the trigger and listbox slots.